policies/recipes/recipedb.go

```diff
@@ -89,8 +89,14 @@ func (db RecipeDB) addRecipe(name, version string, success bool) error {
 	if err != nil {
 		return err
 	}
-	defer f.Close()
-	if _, err = f.Write(dbBytes); err != nil {
+
+	_, err = f.Write(dbBytes)
```

Comment:
Avoid shadowing where you can:

if err := ...

Reply:
is it desirable to shadow variables? My instinct is to not shadow variables but I'll do whatever is most idiomatic in go

Reply:
For most things no, but for err re instantiating for the if block scope is desirable as you wont leak it outside of the scope. This is all because error checks are always if err != nil, so keeping it scoped where possible is preferable.

Reply:
That makes a lot of sense. I changed the error handling style to be more consistent